### PR TITLE
Fixes #4136 Reset all logical switch fields on change of function type

### LINKED ISSUE #4136: Companion logical switch parameters not resetting when function changed
Some parameter values are reset but not all and it depends on what the function is changed from and to.
Even when the function is changed to "----" all parameters are not reset to defaults.
Some old values are hidden depending on the active function.

Companion 2.1.9 Ubuntu 14.04

### PR DESCRIPTION
**Pull request by elecpower**
Port to next of #4136 to reset all logical switch fields on change of function family

### CODE REVIEW

**Comment by elecpower** [comment]
Refer #4517 for merge to master

File: radio/src/gui/128x64/model_logical_switches.cpp [hunk 1]
```diff
@@ -445,7 +445,12 @@ void menuModelLogicalSwitches(event_t event)
           CHECK_INCDEC_MODELVAR_ZERO(event, cs->func, LS_FUNC_MAX);
           uint8_t new_cstate = lswFamily(cs->func);
           if (cstate != new_cstate) {
-            cs->v1 = cs->v2 = (new_cstate==LS_FAMILY_TIMER ? -119/*1.0*/ : 0);
+            unsigned int save_func = cs->func;
+            memset(cs, 0, sizeof(LogicalSwitchData));
+            cs->func = save_func;
+            if (new_cstate == LS_FAMILY_TIMER) {
+              cs->v1 = cs->v2 = -119;
```
Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: You removed the only comment I wrote in this project!

File: radio/src/gui/128x64/model_logical_switches.cpp [hunk 2]
```diff
@@ -445,7 +445,12 @@ void menuModelLogicalSwitches(event_t event)
           CHECK_INCDEC_MODELVAR_ZERO(event, cs->func, LS_FUNC_MAX);
           uint8_t new_cstate = lswFamily(cs->func);
           if (cstate != new_cstate) {
-            cs->v1 = cs->v2 = (new_cstate==LS_FAMILY_TIMER ? -119/*1.0*/ : 0);
+            unsigned int save_func = cs->func;
+            memset(cs, 0, sizeof(LogicalSwitchData));
```
Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: `memclear(...)` saves flash on AVR platforms and does the same job

Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: I would prefer this memclear is called only on ARM and CPUM2560 platforms, for 9X / 9XR I would prefer we cast to uint16_t the structure (or add a new union type) and apply a mask. I will give it a try

Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: This solution costs only 4bytes on 9X:
```
          if (cstate != new_cstate) {
#if defined(CPUARM)
            unsigned int save_func = cs->func;
            memset(cs, 0, sizeof(LogicalSwitchData));
            cs->func = save_func;
#else
            cs->v1 = cs->v2 = 0;
            cs->andsw = 0;
#endif
            if (new_cstate == LS_FAMILY_TIMER) {
              cs->v1 = cs->v2 = -119;
            }
          }
```

**Comment by bsongis** [comment]
I just pushed my proposal above in the branch 'next'

**Pull request closed without merging**